File: src/atcoder/abc310/abc310_a.scala
import java.util.Scanner
import scala.collection.mutable.Seq
import scala.math.min

object Main extends App {
  val sc = new Scanner(System.in)

  val n = sc.nextInt
  val p = sc.nextInt
  val q = sc.nextInt
  val d = Seq.fill(n) { sc.nextInt }
  
  val ans = min(p, q + d.min)
  println(ans)
  
}

File: src/atcoder/abc310/abc310_a.swift
let args = readLine()!.split(separator: " ").map { Int($0)! }
let n = args[0]
let p = args[1]
let q = args[2]
let d = readLine()!.split(separator: " ").map { Int($0)! }
let ans = min(p, q + d.min()!)
print(ans)
